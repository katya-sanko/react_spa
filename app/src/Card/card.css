.card-wrapper {
    width: 33%;
    list-style: none;
}

.card {
    padding: 10px;
}

.card-inlineInfo {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
}

.card-title {
    text-transform: uppercase;
    font-weight: 600;
}

.card-releaseYear {
    border: 1px solid grey;
    font-size: 13px;
    padding: 2px;
    margin-bottom: auto;
}

.card-posterImage {
    width: 100%;
    padding-bottom: 10px
}

.card-genre {
    color: grey;
    font-size: 13px;
}