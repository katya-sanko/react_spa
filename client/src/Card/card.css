.card {
    
}

.card-poster {
    object-fit: contain;
}