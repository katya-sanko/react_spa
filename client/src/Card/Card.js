import React from 'react';
import './card.css';

const Card = ({ movie }) => (
    <div class="card">
        <img class="card-poster" src={movie.poster_path}/>
        {
            movie.genres && movie.genres.map((genre, index) => <span key={index}> {genre}  </span>)
        }
    </div>
);

export default Card;