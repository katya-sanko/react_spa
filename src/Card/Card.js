import React from 'react';
import './card.css';
import { Link } from 'react-router-dom';
import { getReleaseYear } from '../helpers.js';

function getUniqLink(id) {
    return `/movies/film/${id}`;
}

const Card = ({ movie }) => (
    <div className="card">
        <Link to={getUniqLink(movie.id)} className="card-poster">
            <img className="card-posterImage" src={movie.poster_path} />
        </Link>
        <div className="card-inlineInfo">
            <span className="card-title">{movie.title}</span>
            <span className="card-releaseYear">{getReleaseYear(movie.release_date)}</span>
        </div>
        {
            movie.genres && movie.genres.map((genre, index) => <span className="card-genre" key={index}> {genre}  </span>)
        }
    </div>
);

export default Card;