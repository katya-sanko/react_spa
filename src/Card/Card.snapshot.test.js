import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

let mockValue = {
    "id": 299536,
    "title": "Avengers: Infinity War",
    "tagline": "An entire universe. Once and for all.",
    "vote_average": 0,
    "vote_count": 0,
    "release_date": "2018-04-25",
    "poster_path": "https://image.tmdb.org/t/p/w500/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
    "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
    "budget": 500000000,
    "revenue": 0,
    "genres": [
        "Adventure",
        "Science Fiction",
        "Fantasy",
        "Action"
    ],
    "runtime": 156
};

it('renders correctly', () => {
    const tree = renderer
        .create(<Card movie={mockValue} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});