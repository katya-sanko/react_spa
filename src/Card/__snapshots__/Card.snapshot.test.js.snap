// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-poster"
  >
    <img
      className="card-posterImage"
      src="https://image.tmdb.org/t/p/w500/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg"
    />
  </div>
  <div
    className="card-inlineInfo"
  >
    <span
      className="card-title"
    >
      Avengers: Infinity War
    </span>
    <span
      className="card-releaseYear"
    >
      2018
    </span>
  </div>
  <span
    className="card-genre"
  >
     
    Adventure
      
  </span>
  <span
    className="card-genre"
  >
     
    Science Fiction
      
  </span>
  <span
    className="card-genre"
  >
     
    Fantasy
      
  </span>
  <span
    className="card-genre"
  >
     
    Action
      
  </span>
</div>
`;
